### PR TITLE
planner: make queries with the extra column `_tidb_rowid` can use PointGet

### LINKED ISSUE #31543: point_get is not chosen as expected
## Enhancement
schema
``` sql
CREATE TABLE `t` (
  `a` int(11) DEFAULT NULL,
  `b` int(11) DEFAULT NULL,
  UNIQUE KEY `idx` (`a`)
);
```

sql1:
``` sql
explain analyze select t.*, _tidb_rowid from t where a = 1;
```

sql2:
``` sql
explain analyze select t.*  from t where a = 1;
```

Expected:
Both SQL1 and SQL2 use point_get plan.

Actual:
SQL1 does not use the point_get plan.

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #31543

Problem Summary: planner: make queries with the extra column `_tidb_rowid` can use PointGet

### What is changed and how it works?

planner: make queries with the extra column `_tidb_rowid` can use PointGet

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
planner: make queries with the extra column `_tidb_rowid` can use PointGet
```

### CODE REVIEW

**Review by XuHuaiyu — APPROVED** [review]
LGTM

File: planner/core/point_get_plan.go [hunk 1]
```diff
@@ -1476,6 +1476,11 @@ func buildPointDeletePlan(ctx sessionctx.Context, pointPlan PhysicalPlan, dbName
 
 func findCol(tbl *model.TableInfo, colName *ast.ColumnName) *model.ColumnInfo {
 	for _, col := range tbl.Columns {
+		if colName.Name.L == model.ExtraHandleName.L && !tbl.PKIsHandle {
+			colInfo := model.NewExtraHandleColInfo()
+			colInfo.Offset = len(tbl.Columns) - 1
+			return colInfo
+		}
```
Comment (tiancaiamao) [inline planner/core/point_get_plan.go:1483]: This can be moved out of the for loop?

Comment (qw4990) [reply planner/core/point_get_plan.go]: You are right. Updated, PTAL @tiancaiamao

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**